apps/website/src/components/Playground/PlaygroundUrl.ts

```diff
@@ -0,0 +1,50 @@
+import { useActiveCode } from '@codesandbox/sandpack-react';
+import * as lz from 'lz-string';
+import React from 'react';
+
+function useDebounce<T>(value: T, delay: number): T {
+  const [debouncedValue, setDebouncedValue] = React.useState(value);
+
+  React.useEffect(() => {
+    const timeoutId = setTimeout(() => {
+      setDebouncedValue(value);
+    }, delay);
+
+    return () => clearTimeout(timeoutId);
+  }, [value, delay]);
+
+  return debouncedValue;
+}
+
+/**
+ * The component that checks for code modifications in the editor and updates URL to store it code there.
+ */
+export function PlaygroundUrl({ template }: { template: string }) {
+  const { code, updateCode } = useActiveCode();
+  const searchParam = useDebounce(code === template ? null : lz.compressToEncodedURIComponent(code), 400);
```

Comment:
Inspired by TypeScript playground: https://github.com/microsoft/TypeScript-Website/blob/6cc9d3087e5161adc9a0aaef4a2ca18d431501c0/packages/sandbox/src/getInitialCode.ts#L21

Reply:
should it also save to localStorage like ts playground so the user doesn't lose previous code?